tests/shtest

```diff
@@ -207,6 +207,19 @@ fi
 echo '{"a":1,"b",' | $JQ --stream  > /dev/null 2> $d/err
 grep 'Objects must consist of key:value pairs' $d/err > /dev/null
 
+## Regression test for issue #2572 assert when using --jsonargs and invalid JSON
+$JQ -n --jsonargs null invalid && EC=$? || EC=$?
+if [ "$EC" -ne 2 ]; then
+    echo "--jsonargs exited with wrong exit code, expected 2 got $EC" 1>&2
+    exit 1
+fi
+# this tests the args_done code path "--"
+$JQ -n --jsonargs null -- invalid && EC=$? || EC=$?
```

Comment:
Not sure if this should be exit 2 (usage error?) or invalid input error?

I also noticed that we skip using $VALGRIND is some places, should we fix that?

Reply:
We're not very consistent regarding exit status codes.  It'd be nice if a) we documented `jq`'s exit codes besides in the `-e` case, b) maybe made sure that the various usage and system error exit codes differ from the ones for `-e`?.  But that's not for this issue.

And yes, we should fix the non-use of valgrind, but we should do so separately.